pandas/core/strings.py

```diff
@@ -1962,11 +1935,298 @@ def cons_row(x):
+            return (los, fu_wrn)
+        elif isinstance(others, Index):
+            fu_wrn = not others.equals(idx)
+            los = [Series(others.values,
+                          index=(idx if ignore_index else others))]
+            return (los, fu_wrn)
+        elif isinstance(others, DataFrame):
+            fu_wrn = not others.index.equals(idx)
+            if ignore_index and fu_wrn:
+                # without copy, this could change "others"
+                # that was passed to str.cat
+                others = others.copy()
+                others.index = idx
+            return ([others[x] for x in others], fu_wrn)
+        elif isinstance(others, np.ndarray) and others.ndim == 2:
+            others = DataFrame(others, index=idx)
+            return ([others[x] for x in others], False)
+        elif is_list_like(others):
+            others = list(others)  # ensure iterators do not get read twice etc
+            if all(is_list_like(x) for x in others):
+                los = []
+                fu_wrn = False
+                while others:
+                    nxt = others.pop(0)  # list-like as per check above
+                    # safety for iterators and other non-persistent list-likes
+                    # do not map indexed/typed objects; would lose information
+                    if not isinstance(nxt, (DataFrame, Series,
+                                            Index, np.ndarray)):
+                        nxt = list(nxt)
+
+                    # known types without deep inspection
+                    no_deep = ((isinstance(nxt, np.ndarray) and nxt.ndim == 1)
+                               or isinstance(nxt, (Series, Index)))
+                    # Nested list-likes are forbidden - elements of nxt must be
+                    # strings/NaN/None. Need to robustify NaN-check against
+                    # x in nxt being list-like (otherwise ambiguous boolean)
+                    is_legal = ((no_deep and nxt.dtype == object)
+                                or all((isinstance(x, compat.string_types)
+                                        or (not is_list_like(x) and isnull(x))
+                                        or x is None)
+                                       for x in nxt))
+                    # DataFrame is false positive of is_legal
+                    # because "x in df" returns column names
+                    if not is_legal or isinstance(nxt, DataFrame):
+                        raise TypeError(err_msg)
+
+                    nxt, fwn = self._get_series_list(nxt,
+                                                     ignore_index=ignore_index)
+                    los = los + nxt
+                    fu_wrn = fu_wrn or fwn
+                return (los, fu_wrn)
+            # test if there is a mix of list-like and non-list-like (e.g. str)
+            elif (any(is_list_like(x) for x in others)
+                  and any(not is_list_like(x) for x in others)):
```

Comment:
you can make this simpler by just checking for all is not list like (eg strings) 

anything else will fail thru to the TypeError